lib/Remote/HttpCommandExecutor.php

```diff
@@ -182,10 +182,19 @@ class HttpCommandExecutor implements WebDriverCommandExecutor
      * @param string|null $http_proxy
      * @param int|null $http_proxy_port
      */
-    public function __construct($url, $http_proxy = null, $http_proxy_port = null)
+    public function __construct($url, $http_proxy = null, $http_proxy_port = null, $http_proxy_user = null, $http_proxy_pass = null)
     {
         self::$w3cCompliantCommands = array_merge(self::$commands, self::$w3cCompliantCommands);
 
+        $uri = parse_url($http_proxy);
+        if ($uri !== false) {
+            $http_proxy = $uri['host'];
+            $http_proxy_port = $uri['port'];
+            $http_proxy_user = $uri['user'];
+            $http_proxy_pass = $uri['pass'];
+        }
+        unsewt($uri);
```

Comment:
I would remove this whole part of parsing URI. If someone needs this (which would be very uncommon case), they can do this parsing themselves and provide the values directly.